### PR TITLE
Update arange to be (start, stop, step)

### PR DESCRIPTION
**Pull request by madt2709**

### CODE REVIEW

**Comment by sekstini** [comment]
The most common case is to only pass stop, so requiring stop as a keyword argument seems wrong. We should match what PyTorch and Numpy are doing imo.

**Comment by wozeparrot** [comment]
hmm, this makes the base case of `arange` a bit more annoying to use/verbose.

Do we maybe want to overload the base case? I think torch does this.

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/nn/__init__.py                    1               1
tinygrad/tensor.py                         4               2
------------------------------------------------------------
total                                      5               3
------------------------------------------------------------
lines added in the tinygrad folder: 2
```

**Comment by sekstini** [comment]
Here's a simplified version of how PyTorch handles it:

```python
def arange(start=0, end=None, step=1):
  if end is None:
    end = start
    start = 0
  ...
```

**Comment by madt2709** [comment]
Ah thanks, I knew there had to be a smarter way! will update now

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -152,7 +152,11 @@ def zeros(*shape, **kwargs): return Tensor.full(argfix(*shape), 0, **kwargs)
   def ones(*shape, **kwargs): return Tensor.full(argfix(*shape), 1, **kwargs)
 
   @staticmethod
-  def arange(stop, start=0, step=1, **kwargs): return Tensor.full((ceil((stop-start)/step),), step, **kwargs).cumsum() + (start - step)
+  def arange(start=0, stop=None, step=1, **kwargs):
+    if stop is None:
+      stop = start
+      start = 0
```
Comment (wozeparrot) [inline tinygrad/tensor.py:158]: Could use `if stop is None: stop, start = start, 0`

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -152,7 +152,11 @@ def zeros(*shape, **kwargs): return Tensor.full(argfix(*shape), 0, **kwargs)
   def ones(*shape, **kwargs): return Tensor.full(argfix(*shape), 1, **kwargs)
 
   @staticmethod
-  def arange(stop, start=0, step=1, **kwargs): return Tensor.full((ceil((stop-start)/step),), step, **kwargs).cumsum() + (start - step)
+  def arange(start=0, stop=None, step=1, **kwargs):
```
Comment (wozeparrot) [inline tinygrad/tensor.py:155]: Can we make `start` required?

Comment (madt2709) [reply tinygrad/tensor.py:155]: ah yes good point. 

(sorry being really slow on this pr)

File: tinygrad/tensor.py (hunk 1 above)
Comment (madt2709) [reply tinygrad/tensor.py:158]: sure, less lines for sure

**Comment by wozeparrot** [comment]
Much better, will merge if tests pass.

**Pull request merged**